src/Action/SyliusConvertAction.php

```diff
@@ -101,7 +123,8 @@ protected function setReference(ArrayObject $model, PaymentInterface $payment):
     {
         // The ID should be always unique so we can use it,
         // but we can also use Unix timestamp to get a really uniq value
-        $model['reference'] = sprintf(static::PAYMENT_ID_FORMAT, $payment->getId());
+//        $model['reference'] = sprintf(static::PAYMENT_ID_FORMAT, $payment->getId());
+        $model['reference'] = substr(uniqid(), 0, 12);
```

Comment:
This lines can be delegated to one of `Sylius\Component\Core\Payment\InvoiceNumberGeneratorInterface` depending services like into the PayPal Express Checkout : https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PayumBundle/Action/Paypal/ExpressCheckout/ConvertPaymentAction.php#L49